AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -991,6 +1026,90 @@ - (CGRect)convertRect:(CGRect)rect toNode:(ASDisplayNode *)node
   return CGRectApplyAffineTransform(rect, flattenedTransform);
 }
 
+#pragma mark - Layout Transition
+
+- (void)transitionLayoutWithAnimation:(BOOL)animated
+{
+  [self transitionLayoutThatFits:_constrainedSize animated:animated];
+}
+
+- (void)transitionLayoutThatFits:(ASSizeRange)constrainedSize animated:(BOOL)animated
+{
+  [self invalidateCalculatedLayout];
+  _deferImmediateHierarchyManagement = YES;
+  [self measureWithSizeRange:constrainedSize]; // Generate a new layout
+  _deferImmediateHierarchyManagement = NO;
+  [self __transitionLayoutWithAnimation:animated];
+}
+
+- (void)__transitionLayoutWithAnimation:(BOOL)animated
+{
+  _transitionContext = [[_ASTransitionContext alloc] initWithAnimation:animated delegate:self];
+  [self __implicitlyInsertSubnodes];
+  [self animateLayoutTransition:_transitionContext];
+}
+
+- (void)animateLayoutTransition:(id<ASContextTransitioning>)context
+{
+  [self __layoutSublayouts];
+  [context completeTransition:YES];
+}
+
+- (void)didCompleteTransitionLayout:(id<ASContextTransitioning>)context
+{
+  [self __implicitlyRemoveSubnodes];
+}
+
+#pragma mark - Implicit node hierarchy managagment
+
+- (void)__implicitlyInsertSubnodes
+{
+  if ([_insertedSubnodes count]) {
+    for (_ASDisplayNodePosition *position in _insertedSubnodes) {
+      [self insertSubnode:position.node atIndex:position.index];
+    }
+    _insertedSubnodes = nil;
+  }
+}
+
+- (void)__implicitlyRemoveSubnodes
+{
+  if ([_deletedSubnodes count]) {
```

Comment:
For these two methods, in their final form is it possible to remove the if clause?